File: restdoc-http-client/src/main/java/restdoc/client/EchoController.java
package restdoc.client;

import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;

import java.util.Date;
import java.util.HashMap;

@RestController
public class EchoController {

    @GetMapping(value = "/echo/{var}", produces = MediaType.APPLICATION_JSON_VALUE)
    Object echo(@PathVariable String var
    ) {


        System.err.println(var);
        HashMap<Object, Object> map = new HashMap<>();
        map.put("serviceName", "restdoc-starter");
        map.put("time", new Date().getTime());
        map.put("success", true);
        map.put("system", "windows");
        map.put("tcp model", "nio channel");
        map.put("tcp framework", "netty-4.0.Final");
        return map;
    }
}

File: restdoc-rpc-client/restdoc-restweb-client/src/main/java/restdoc/client/EchoController.kt
package restdoc.client

import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RestController
import java.util.*

@RestController
class EchoController {
    @GetMapping(value = ["/echo/{var}"], produces = [MediaType.APPLICATION_JSON_VALUE]) fun echo(@PathVariable `var`: String?
    ): Any {
        System.err.println(`var`)
        val map = HashMap<Any, Any>()
        map["serviceName"] = "restdoc-starter"
        map["time"] = Date().time
        map["success"] = true
        map["system"] = "windows"
        map["tcp model"] = "nio channel"
        map["tcp framework"] = "netty-4.0.Final"
        return map
    }
}